### PR TITLE
chore: add initial implementation and documentation

### PR DESCRIPTION
**Pull request by DmitriyMV**
Move things from Talos and other projects.

Signed-off-by: Dmitriy Matrenichev <dmitry.matrenichev@siderolabs.com>

### CODE REVIEW

File: util/util.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package util contains utility functions and types.
+package util
```
Comment (smira) [inline util/util.go]: we could name the package `value`, so it's `value.IsZero(x)`

**Review by smira — APPROVED** [review]

File: util/util.go (hunk 1 above)
Comment (DmitriyMV) [reply util/util.go]: Agree. Moment.

Comment (DmitriyMV) [reply util/util.go]: Done!

File: value/util.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (smira) [inline value/util.go]: nit: `s/util.go/value.go/`

Comment (DmitriyMV) [reply value/util.go]: Thanks, forgot!

**Pull request merged**